File: pertemuan11/PR/PR11_2473021_JAVA/B_HitungLuas.java

/**
 * Pertemuan 11
 * [PR]  program untuk menghitung luas dan keliling
 *
 * @author 2473021-Febrianus Leona Putra
 * @version 06 Desember 2024
 */
import java.util.Scanner;

public class B_HitungLuas {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int pilih = 0;

        while (true) {
            System.out.println("\nPilih Luas");
            System.out.println("1. Luas Persegi");
            System.out.println("2. Luas Persegi Panjang");
            System.out.println("3. Luas Lingkaran");
            System.out.println("4. Luas Jajargenjang");

            System.out.print("Pilihan: ");
            String input = sc.nextLine();

            if (isInteger(input)) {
                pilih = Integer.parseInt(input);
                if (pilih >= 1 && pilih <= 4) {
                    break;
                } else {
                    System.out.println("Input tidak valid");
                }
            } else {
                System.out.println("Input tidak valid");
            }
        }

        switch (pilih) {
            case 1:
                luasPersegi(sc);
                break;
            case 2:
                luasPP(sc);
                break;
            case 3:
                luasLingkaran(sc);
                break;
            case 4:
                luasJajargenjang(sc);
                break;
        }

        sc.close();
    }

    private static boolean isInteger(String str) {
        for (char c : str.toCharArray()) {
            if (!Character.isDigit(c)) {
                return false;
            }
        }
        return !str.isEmpty();
    }

    private static void luasPersegi(Scanner sc) {
        System.out.print("Input Sisi: ");
        int sisi = Integer.parseInt(sc.nextLine());
        int luas = sisi * sisi;
        int keliling = 4 * sisi;
        System.out.println("Hasil Luas: " + luas);
        System.out.println("Hasil Keliling: " + keliling);
    }

    private static void luasPP(Scanner sc) {
        System.out.print("Input Panjang: ");
        int panjang = Integer.parseInt(sc.nextLine());
        System.out.print("Input Lebar: ");
        int lebar = Integer.parseInt(sc.nextLine());
        int luas = panjang * lebar;
        int keliling = 2 * (panjang + lebar);
        System.out.println("Hasil Luas: " + luas);
        System.out.println("Hasil Keliling: " + keliling);
    }

    private static void luasLingkaran(Scanner sc) {
        System.out.print("Input Jari-jari: ");
        int jariJari = Integer.parseInt(sc.nextLine());
        double luas = 3.14 * jariJari * jariJari;
        double keliling = 2 * 3.14 * jariJari;
        System.out.println("Hasil Luas: " + luas);
        System.out.println("Hasil Keliling: " + keliling);
    }

    private static void luasJajargenjang(Scanner sc) {
        System.out.print("Input Alas: ");
        int alas = Integer.parseInt(sc.nextLine());
        System.out.print("Input Tinggi: ");
        int tinggi = Integer.parseInt(sc.nextLine());
        System.out.print("Input Sisi Miring: ");
        int sisiMiring = Integer.parseInt(sc.nextLine());
        int luas = alas * tinggi;
        int keliling = 2 * (alas + sisiMiring);
        System.out.println("Hasil Luas: " + luas);
        System.out.println("Hasil Keliling: " + keliling);
    }
}

File: pertemuan11/PR/PR11_2473021_PY/B_HitungLuas.py
"""
Pertemuan 11
[PR] program untuk menghitung luas dan keliling

@author 2473021-Febrianus Leona Putra
@version 06 Desember 2024
"""
def luas_persegi():
    sisi = int(input("Input Sisi: "))
    luas = sisi * sisi
    keliling = 4 * sisi
    print(f"Hasil Luas: {luas}")
    print(f"Hasil Keliling: {keliling}")

def luas_pp():
    panjang = int(input("Input Panjang: "))
    lebar = int(input("Input Lebar: "))
    luas = panjang * lebar
    keliling = 2 * (panjang + lebar)
    print(f"Hasil Luas: {luas}")
    print(f"Hasil Keliling: {keliling}")

def luas_lingkaran():
    jari_jari = int(input("Input Jari-jari: "))
    luas = 3.14 * (jari_jari ** 2)
    keliling = 2 * 3.14 * jari_jari
    print(f"Hasil Luas: {luas}")
    print(f"Hasil Keliling: {keliling}")

def luas_jajargenjang():
    alas = int(input("Input Alas: "))
    tinggi = int(input("Input Tinggi: "))
    sisi_miring = int(input("Input Sisi Miring: "))
    luas = alas * tinggi
    keliling = 2 * (alas + sisi_miring)
    print(f"Hasil Luas: {luas}")
    print(f"Hasil Keliling: {keliling}")

while True:
    print("\nPilih Luas")
    print("1. Luas Persegi")
    print("2. Luas Persegi Panjang")
    print("3. Luas Lingkaran")
    print("4. Luas Jajargenjang")

    pilih = input("Pilihan: ")

    if pilih.isdigit():
        pilih = int(pilih)
        if 1 <= pilih <= 4:
            break
        else:
            print("Pilihan tidak valid")
    else:
        print("Input tidak valid")

if pilih == 1:
    luas_persegi()
elif pilih == 2:
    luas_pp()
elif pilih == 3:
    luas_lingkaran()
elif pilih == 4:
    luas_jajargenjang()
